src/main/java/org/opensearch/security/privileges/PrivilegesEvaluator.java

```diff
@@ -226,15 +224,15 @@ public PrivilegesEvaluator(
 
     void updateConfiguration(
         SecurityDynamicConfiguration<ActionGroupsV7> actionGroupsConfiguration,
-        SecurityDynamicConfiguration<RoleV7> rolesConfiguration
+        SecurityDynamicConfiguration<RoleV7> rolesConfiguration,
+        SecurityDynamicConfiguration<TenantV7> tenantConfiguration
     ) {
-        if (rolesConfiguration != null) {
-            SecurityDynamicConfiguration<ActionGroupsV7> actionGroupsWithStatics = actionGroupsConfiguration != null
-                ? DynamicConfigFactory.addStatics(actionGroupsConfiguration.clone())
-                : DynamicConfigFactory.addStatics(SecurityDynamicConfiguration.empty(CType.ACTIONGROUPS));
-            FlattenedActionGroups flattenedActionGroups = new FlattenedActionGroups(actionGroupsWithStatics);
+        FlattenedActionGroups flattenedActionGroups = new FlattenedActionGroups(actionGroupsConfiguration.withStaticConfig());
+        rolesConfiguration = rolesConfiguration.withStaticConfig();
+        tenantConfiguration = tenantConfiguration.withStaticConfig();
+        try {
```

Comment:
why is this `try { } catch { }` required now?

Reply:
It's basically just this try/catch block moved down one level and split up for action privileges and tenant privileges:

https://github.com/opensearch-project/security/blob/344673a455de956f6a8f3217e61d0636b46a3527/src/main/java/org/opensearch/security/privileges/PrivilegesEvaluator.java#L203-L212

Indeed, one could argue whether we should have try/catch blocks here at all. I do not really have a strong opinion on it.